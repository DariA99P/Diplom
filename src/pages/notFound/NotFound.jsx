import React from 'react';
import { Card } from 'antd';
import './style.css';

const NotFound = () => (
  <Card>
    <h1>Not Found</h1>
    <p>Sorry, the page you visited does not exist or is not available.</p>
    <p>Contact your administrator if you are having problems.</p>
  </Card>
);

export default NotFound;
