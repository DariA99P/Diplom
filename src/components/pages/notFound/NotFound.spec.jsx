import React from 'react';
import { shallow } from 'enzyme';

import NotFound from '.';

describe('<NotFound />', () => {
  describe('When rendering', () => {
    const tree = shallow(<NotFound />);

    it('should renders properly', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe('When back button is clicked', () => {
    const goBack = jest.fn();
    const history = { goBack };
    const wrapper = shallow(<NotFound history={history} />);

    it('should redirects to back page', () => {
      wrapper.find('Button').simulate('click');
      expect(goBack).toMatchSnapshot();
    });
  });
});
