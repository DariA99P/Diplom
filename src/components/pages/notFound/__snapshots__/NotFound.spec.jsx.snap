// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> When back button is clicked should redirects to back page 1`] = `[Function]`;

exports[`<NotFound /> When rendering should renders properly 1`] = `
<Row
  align="middle"
  gutter={76}
  justify="center"
  type="flex"
>
  <Col
    lg={8}
    md={24}
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <img
      alt="Not Found"
      className="img"
      src="not_found.png"
    />
  </Col>
  <Col
    lg={8}
    md={24}
  >
    <h1
      className="title"
    >
      Error
    </h1>
    <p
      className="text"
    >
      Sorry, the page you visited does not exist or is not available. Contact your administrator if you are having problems.
    </p>
    <Button
      ghost={false}
      loading={false}
      onClick={[Function]}
      prefixCls="ant-btn"
      type="primary"
    >
      Back
    </Button>
  </Col>
</Row>
`;
